frmts/zarr/zarr_v3_codec.cpp

```diff
@@ -29,18 +29,76 @@ ZarrV3Codec::ZarrV3Codec(const std::string &osName) : m_osName(osName)
 ZarrV3Codec::~ZarrV3Codec() = default;
 
 /************************************************************************/
-/*                        ZarrV3CodecGZip()                             */
+/*                      ZarrV3CodecAbstractCompressor()                 */
+/************************************************************************/
+
+ZarrV3CodecAbstractCompressor::ZarrV3CodecAbstractCompressor(
+    const std::string &osName)
+    : ZarrV3Codec(osName)
+{
+}
+
+/************************************************************************/
+/*                 ZarrV3CodecAbstractCompressor::Encode()              */
+/************************************************************************/
+
+bool ZarrV3CodecAbstractCompressor::Encode(
+    const ZarrByteVectorQuickResize &abySrc,
+    ZarrByteVectorQuickResize &abyDst) const
+{
+    abyDst.resize(abyDst.capacity());
+    void *pOutputData = abyDst.data();
+    size_t nOutputSize = abyDst.size();
+    bool bRet = m_pCompressor->pfnFunc(
+        abySrc.data(), abySrc.size(), &pOutputData, &nOutputSize,
+        m_aosCompressorOptions.List(), m_pCompressor->user_data);
+    if (bRet)
+    {
+        abyDst.resize(nOutputSize);
+    }
+    else if (nOutputSize > abyDst.size())
+    {
+        CPLError(CE_Failure, CPLE_AppDefined,
+                 "%s codec:Encode(): output buffer too small",
```

Comment:
I know we are not translating error messages but I really think we should eventually. 
Can we introduce _() at least in new code?

Reply:
cf my comment in another ticket. I believe this should be part of a more general move, if we eventually go into that direction.  A more practical reason is that the _() macro isn't in any of our headers currently, and we should be cautious how/where to introduce it to avoid toe-stepping over gettext...